frontend/src/components/NavBar.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../hooks/authContext";
+import "./NavBar.css";
+import LoginIcon from "../assets/login-icon.svg"; // Import your custom icon
+
+const Navbar = () => {
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [cyclesDropdownOpen, setCyclesDropdownOpen] = useState(false);
+  const { user, logout } = useAuth();
+  const navigate = useNavigate();
+  const [scrollToNews, setScrollToNews] = useState(false); // Track NEWS click intent
+
+  const handleAccountClick = () => {
+    if (user) {
+      navigate("/dashboard");
+    } else {
+      navigate("/account");
+    }
+    setMenuOpen(false);
+  };
+
+  const handleLogout = () => {
+    logout();
+    navigate("/");
+    setMenuOpen(false);
+  };
+
+  const toggleMenu = () => {
+    setMenuOpen(!menuOpen);
+  };
+
+  const toggleCyclesDropdown = () => {
+    setCyclesDropdownOpen(!cyclesDropdownOpen);
+  };
+
+  // 🔹 Scroll to News Section (Navigate First If Needed)
+  const handleScrollToNews = () => {
+    if (window.location.pathname !== "/") {
+      setScrollToNews(true); // Mark intent to scroll
+      navigate("/"); // Navigate to home
+    } else {
+      scrollToNewsSection(); // Scroll immediately if already on home
+    }
+    setMenuOpen(false);
+  };
+
+  // 🔹 Scroll to News Section
+  const scrollToNewsSection = () => {
+    const newsSection = document.getElementById("news-section");
+    if (newsSection) {
+      newsSection.scrollIntoView({ behavior: "smooth" });
+      setScrollToNews(false); // Reset intent after scrolling
+    }
+  };
+
+  // 🔹 After navigation, scroll to NEWS if needed
+  useEffect(() => {
+    if (scrollToNews && window.location.pathname === "/") {
+      setTimeout(scrollToNewsSection, 500); // Wait to ensure page is loaded before scrolling
+    }
+  }, [scrollToNews, window.location.pathname]);
+
+  return (
+    <header className="navbar">
+      <div className="hamburger" onClick={toggleMenu}>
+        <span className={`line ${menuOpen ? "open" : ""}`}></span>
+        <span className={`line ${menuOpen ? "open" : ""}`}></span>
+        <span className={`line ${menuOpen ? "open" : ""}`}></span>
+      </div>
+
```

Comment:
this feels a little overkill, better to put the classname on the whole hamburger instead of all lines. However, I cant see that the class open is even used there?